packages/nimbus/src/components/number-input/number-input.tsx

```diff
@@ -58,12 +58,18 @@ export const NumberInput = (props: NumberInputProps) => {
   };
 
   return (
-    <NumberInputRootSlot {...recipeProps} {...styleProps} size={size}>
+    <NumberInputRootSlot
+      {...recipeProps}
+      {...styleProps}
+      size={size}
+      className={props?.className as string}
+    >
       <NumberInputInputSlot
         ref={ref}
         {...inputProps}
         {...stateProps}
-        size={size}
+        // https://github.com/adobe/react-spectrum/issues/4744
+        name={props.name}
```

Comment:
This is for backwards-compat with UI kit, and also because our forms use `formik` so we pass the right value back on submit anyway